ouroboros-consensus/src/Ouroboros/Consensus/Mempool/API.hs

```diff
@@ -85,14 +85,44 @@ data Mempool m blk idx = Mempool {
       -- @STM m@; we keep it in @m@ instead to leave open the possibility of
       -- persistence.
       --
-      -- The transactions will be validated, /in order/, before they are added
-      -- to the mempool. Invalid transactions will be returned along with the
-      -- validation error. In principle it is possible that such validation
-      -- errors are transient; for example, it is possible that a transaction is
-      -- rejected because one of its inputs is not /yet/ available in the UTxO
-      -- (the transaction it depends on is not yet in the chain, nor in the
-      -- mempool). In practice however it is likely that rejected transactions
-      -- will still be rejected later, and should just be dropped.
+      -- The following validation steps will be performed when adding
+      -- transactions to the mempool:
+      --
+      -- * Transactions which already exist in the mempool are revalidated,
+      --   /in order/, against the current ledger state. Existing transactions
+      --   which are found to be invalid, with respect to the current ledger
+      --   state, are dropped from the mempool, whereas valid transactions
+      --   remain in the mempool.
+      -- * The new transactions provided will be validated, /in order/,
+      --   against the current ledger state. Transactions which are found to
+      --   be invalid, with respect to the current ledger state, are dropped,
+      --   whereas valid transactions are added to the mempool.
+      --
+      -- Note that transactions that are invalid, with respect to the current
+      -- ledger state, will /never/ be added to the mempool. However, it is
+      -- possible that, at a given point in time, transactions which were once
+      -- valid but are now invalid, with respect to the current ledger state,
+      -- could exist within the mempool until they are revalidated and dropped
+      -- from the mempool via a call to either 'addTxs' or 'syncState'.
+      --
+      -- This function will return a list potentially consisting of a few
+      -- different kinds of transactions:
+      --
+      -- * Those transactions provided which were found to be valid, along
+      --   with 'Nothing' for their accompanying @Maybe (ApplyTxErr blk)@
+      --   values.
+      -- * Those transactions provided which were found to be invalid, along
+      --   with their accompanying validation errors.
+      -- * Invalid transactions, with respect to the current ledger state,
```

Comment:
This might change, see https://github.com/input-output-hk/ouroboros-network/pull/696#discussion_r300960401